api/features/models.py

```diff
@@ -406,6 +409,35 @@ def get_delete_log_message(self, history_instance) -> typing.Optional[str]:
     def _get_environment(self) -> "Environment":
         return self.environment
 
+    @hook(AFTER_DELETE)
+    def create_github_comment(self) -> None:
+        from integrations.github.dataclasses import GithubData
+        from integrations.github.github import generate_data
+        from integrations.github.tasks import (
+            call_github_app_webhook_for_feature_state,
+        )
+        from webhooks.webhooks import WebhookEventType
+
+        if (
+            self.feature.external_resources.exists()
+            and self.feature.project.github_project.exists()
+            and self.feature.project.organisation.github_config.exists()
+        ):
+            github_configuration = GithubConfiguration.objects.get(
+                organisation_id=self.feature.project.organisation_id
+            )
+
+            feature_data: GithubData = generate_data(
+                github_configuration=github_configuration,
+                feature=self.feature,
+                type=WebhookEventType.SEGMENT_OVERRIDE_DELETED.value,
+                segment_name=self.segment.name,
+            )
+
+            call_github_app_webhook_for_feature_state.delay(
+                args=(asdict(feature_data),),
+            )
```

Comment:
I'd rather this code belonged in `integrations.github` somewhere (the amount of imports alone suggests it).

Reply:
Looking at both this code and the one in `api/features/versioning/serializers.py`, I'm pretty sure it can be factored out to a single interface.

Reply:
I created a common function to handle this code.